src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessCompletionOfNextDataSetVersionImportFunctionTests.cs

```diff
@@ -117,6 +124,3168 @@ private static Mock<TaskOrchestrationContext> DefaultMockOrchestrationContext(Gu
+                change: filterOptionMetaChanges[3]);
+            AssertFilterOptionChanged(
+                expectedOldOptionLink: oldFilterMetas[SqidEncoder.Encode(2)][SqidEncoder.Encode(7)],
+                expectedNewOptionLink: newFilterMetas[SqidEncoder.Encode(2)][SqidEncoder.Encode(7)],
+                change: filterOptionMetaChanges[4]);
+            AssertFilterOptionDeleted(
+                expectedOptionLink: oldFilterMetas[SqidEncoder.Encode(2)][SqidEncoder.Encode(6)],
+                change: filterOptionMetaChanges[5]);
+            AssertFilterOptionChanged(
+                expectedOldOptionLink: oldFilterMetas[SqidEncoder.Encode(2)][SqidEncoder.Encode(8)],
+                expectedNewOptionLink: newFilterMetas[SqidEncoder.Encode(2)][SqidEncoder.Encode(8)],
+                change: filterOptionMetaChanges[6]);
+            AssertFilterOptionChanged(
+                expectedOldOptionLink: oldFilterMetas[SqidEncoder.Encode(1)][SqidEncoder.Encode(4)],
+                expectedNewOptionLink: newFilterMetas[SqidEncoder.Encode(1)][SqidEncoder.Encode(4)],
+                change: filterOptionMetaChanges[7]);
+            AssertFilterOptionAdded(
+                expectedOptionLink: newFilterMetas[SqidEncoder.Encode(1)][SqidEncoder.Encode(10)],
+                change: filterOptionMetaChanges[8]);
+            AssertFilterOptionAdded(
+                expectedOptionLink: newFilterMetas[SqidEncoder.Encode(2)][SqidEncoder.Encode(11)],
+                change: filterOptionMetaChanges[9]);
+            AssertFilterOptionAdded(
+                expectedOptionLink: newFilterMetas[SqidEncoder.Encode(1)][SqidEncoder.Encode(9)],
+                change: filterOptionMetaChanges[10]);
+            AssertFilterOptionAdded(
+                expectedOptionLink: newFilterMetas[SqidEncoder.Encode(2)][SqidEncoder.Encode(12)],
+                change: filterOptionMetaChanges[11]);
+        }
+
+        private static void AssertSingleFilterDeleted(
+            IReadOnlyList<FilterMetaChange> changes,
+            FilterMeta expectedFilterMeta)
+        {
+            Assert.Single(changes,
+                c => c.PreviousStateId == expectedFilterMeta.Id
+                     && c.CurrentStateId is null);
+        }
+
+        private static void AssertSingleFilterAdded(
+            IReadOnlyList<FilterMetaChange> changes,
+            FilterMeta expectedFilterMeta)
+        {
+            Assert.Single(changes,
+                c => c.PreviousStateId is null
+                     && c.CurrentStateId == expectedFilterMeta.Id);
+        }
+
+        private static void AssertSingleFilterChanged(
+            IReadOnlyList<FilterMetaChange> changes,
+            FilterMeta expectedOldFilterMeta,
+            FilterMeta expectedNewFilterMeta)
+        {
+            Assert.Single(changes,
+                c => c.PreviousStateId == expectedOldFilterMeta.Id
+                     && c.CurrentStateId == expectedNewFilterMeta.Id);
+        }
```

Comment:
Yes, so these are here so that I don't have to worry about the **order** of the changes when I'm asserting in the majority of the tests. The order for those tests is unimportant and extra overhead to think about - hence, the 2 different sets of private methods. And then I use the other private methods inside the tests for the **ordering**.

Reply:
I'm not sure why we wouldn't just test the ordering at the same time though. It's not a lot more effort, it gives us more reassurance that ordering is correct in multiple places and we have to maintain less code (which IMO is the biggy here).

Reply:
Because then the test has 2 responsibilities, and I was trying to spin out new tests for new (independent) things I wanted to test. I personally think the ordering makes far more sense as it's own test - we could easily decide in the future that we don't care about the ordering anymore, and we'd only have to remove the ordering test; whilst leaving the other tests as they are. I also just think testing for the ordering was quite thought-intensive, and the extra overhead of doing this in all of the tests seems a bit unnecessary to me. We'd have to spend a lot of time adding the ordering into each test set-up (adding labels to everything etc.) and each test assertions. And, on top of that, the person reading the tests would have to think a lot more when trying to decide 'Yes, I agree that this change should come next because if I scroll up that is the 4th in the list when I alphabetically order them'. I think keeping the ordering tests separate really simplifies things, and paying the price for these other private methods is worth it.